dashpay/src/main/kotlin/org/dashevo/dashpay/ContactRequests.kt

```diff
@@ -1,16 +1,47 @@
 package org.dashevo.dashpay
 
+import org.bitcoinj.core.Base58
+import org.bouncycastle.crypto.params.KeyParameter
+import org.bouncycastle.util.encoders.Base64
 import org.dashevo.dapiclient.model.DocumentQuery
 import org.dashevo.dpp.document.Document
+import org.dashevo.dpp.identity.Identity
 import org.dashevo.platform.Documents
 import org.dashevo.platform.Platform
+import java.util.*
+import kotlin.collections.ArrayList
 
 class ContactRequests(val platform: Platform) {
 
     companion object {
         const val CONTACTREQUEST_DOCUMENT = "dashpay.contactRequest"
     }
 
+    fun create(fromUser: BlockchainIdentity, toUser: Identity, aesKey: KeyParameter) {
+        val contactPubKey = fromUser.getReceiveFromContactChain(toUser, aesKey)
+        val xpub = Base58.decode(contactPubKey.watchingKey.serializePubB58(platform.params))
+        //TODO: Remove line below. We're generating 112 bytes and the contract specifies 96 bytes for this field (128 Base64 chars).
+        val xpubWithoutIV = xpub.copyOfRange(15, xpub.size-1)
```

Comment:
xpub's don't have IV's. The IV comes from `fromUser.encryptExtendedPublicKey` 

I am sure this will be remedied in a future commit.

Reply:
Though, technically, this part doesn't need to work, since this story only needs to create the request and payments are a separate story.

Reply:
This doesn't need to be fixed since the ECDH doesn't match iOS.

Reply:
Instead of this hack which puts junk data into the "encryptedPublicKey" we could just put 128 Zeros.  This would readily identify the field as invalid.

On the other hand, we could leave the junk data which can be used later to create error handling for invalid encrypted data.